### DISCUSSION TITLE
Access to relationship data in the mutateFormDataBeforeSave

### DISCUSSION THREAD

**Discussion by sh4msi** [filamentphp/filament#4574] [Help] [answered]
hi,
How can I access the data of relation fields in mutateFormDataBeforeSave?

```
Repeater::make('bank_accounts')
              ->relationship('bankAccounts')
             ...

Fieldset::make('User Extra Info')
              ->relationship('extraInfo')
             ...
```

**Comment by danharrin** [comment, marked as answer]
Those components have a dedicated `->mutateRelationshipDataBeforeSaveUsing(function (array $data) {})` which you can use to do the same thing

**Reply by sh4msi** [reply]
mutateRelationshipDataBeforeSaveUsing is only for Repeater and Fieldset does not have a method with this name

**Reply by danharrin** [reply]
Make sure you're on the latest update... I'm pretty sure it does

**Reply by sh4msi** [reply]
Thanks, I updated it and it was fixed

How can I access other data entered in the form in mutateRelationshipDataBeforeSaveUsing?

**Reply by danharrin** [reply]
Using `$get` like other field methods
